From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Minh Nguyen <minhh2792.work@gmail.com>
Date: Tue, 3 May 2022 14:38:52 +0000
Subject: [PATCH] Pearl Configuration


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 0eda17c8642aa49c2f57c409c3bd3234948feae9..ba0efc789fa2af0ac50697039ed8d7cc847ae746 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1902,6 +1902,19 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // Pearl start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getPearlConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @NotNull
+        public java.util.Properties getServerProperties() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Pearl end
+
         /**
          * Sends the component to the player
          *
