From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Minh Nguyen <minhh2792.work@gmail.com>
Date: Sun, 13 Mar 2022 10:36:50 +0000
Subject: [PATCH] Always agree to EULA


diff --git a/src/main/java/net/minecraft/server/Eula.java b/src/main/java/net/minecraft/server/Eula.java
index 920a0e6540bc8b8ce34dfaf57d58c200013b7b2c..bc519ce56e78f3ec4811ce23cf39e45e3c70d21f 100644
--- a/src/main/java/net/minecraft/server/Eula.java
+++ b/src/main/java/net/minecraft/server/Eula.java
@@ -16,7 +16,7 @@ public class Eula {
 
     public Eula(Path eulaFile) {
         this.file = eulaFile;
-        this.agreed = SharedConstants.IS_RUNNING_IN_IDE || this.readFile();
+        this.agreed = true; // Pearl - yes we are agreed to the EULA
     }
 
     private boolean readFile() {
