From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 3 Mar 2022 14:19:19 +0700
Subject: [PATCH] EMC: Mark Withers as Persistent

We don't want them despawning due to player range

diff --git a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
index 75f4749c35a4cc35bc8617b1ba5c74ef198a423c..182862266f549e103cd4f7cc34837273c8a93ec6 100644
--- a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -747,6 +747,13 @@ public class WitherBoss extends Monster implements PowerableMob, RangedAttackMob
         return this.getHealth() <= this.getMaxHealth() / 2.0F;
     }
 
+    // EMC start
+    @Override
+    public boolean removeWhenFarAway(double distanceSquared) {
+        return false;
+    }
+    // EMC end
+
     @Override
     public MobType getMobType() {
         return MobType.UNDEAD;
