From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Minh Nguyen <minhh2792.work@gmail.com>
Date: Sat, 16 Apr 2022 01:44:32 +0000
Subject: [PATCH] Configurable map update interval


diff --git a/src/main/java/me/minhh2792/pearl/PearlConfig.java b/src/main/java/me/minhh2792/pearl/PearlConfig.java
index 3befc3e286f46771407b40a20955de1eb7d19572..ca98bdd7df104bcb0e16438fe9f0ff98d3ae7631 100644
--- a/src/main/java/me/minhh2792/pearl/PearlConfig.java
+++ b/src/main/java/me/minhh2792/pearl/PearlConfig.java
@@ -155,6 +155,11 @@ public class PearlConfig {
         config.addDefault(path, def);
         return config.getString(path, config.getString(path));
     }
+
+	public static int mapUpdateInterval = 5;
+	private static void mapUpdateInterval() {
+		mapUpdateInterval = getInt("item.map.update-interval", mapUpdateInterval);
+	}
 	
 
 }
diff --git a/src/main/java/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java b/src/main/java/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
index 6a78374b6ab3198571df7e17023d371667ebb24b..cce9039621dfff63eebb1a0f1aaa474f644d76e1 100644
--- a/src/main/java/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
+++ b/src/main/java/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
@@ -624,7 +624,7 @@ public class MapItemSavedData extends SavedData {
         @Nullable
         Packet<?> nextUpdatePacket(int mapId) {
             MapItemSavedData.MapPatch worldmap_b;
-            if (!this.dirtyData && this.tick % 5 != 0) { this.tick++; return null; } // Paper - this won't end up sending, so don't render it!
+            if (!this.dirtyData && this.tick % me.minhh2792.pearl.PearlConfig.mapUpdateInterval != 0) { this.tick++; return null; } // Paper - this won't end up sending, so don't render it! // Pearl - Configurable map update interval
             boolean vanillaMaps = shouldUseVanillaMap(); // Paper
             org.bukkit.craftbukkit.map.RenderData render = !vanillaMaps ? MapItemSavedData.this.mapView.render((org.bukkit.craftbukkit.entity.CraftPlayer) this.player.getBukkitEntity()) : MapItemSavedData.this.vanillaRender; // CraftBukkit // Paper
 
@@ -637,7 +637,7 @@ public class MapItemSavedData extends SavedData {
 
             Collection collection;
 
-            if ((true || this.dirtyDecorations) && this.tick++ % 5 == 0) { // CraftBukkit - custom maps don't update this yet
+            if ((true || this.dirtyDecorations) && this.tick++ % me.minhh2792.pearl.PearlConfig.mapUpdateInterval == 0) { // CraftBukkit - custom maps don't update this yet // Pearl - Configurable map update interval
                 this.dirtyDecorations = false;
                 // CraftBukkit start
                 java.util.Collection<MapDecoration> icons = new java.util.ArrayList<MapDecoration>();
